backend/src/main/java/capstone/backend/services/OrderToSupplierService.java

```diff
@@ -44,5 +53,35 @@ private boolean checkIfProductsExistent(OrderToSupplierDTO order) {
                 .toList()
                 .contains(false);
     }
+    private boolean supplierDoesNotCarryProduct(OrderToSupplierDTO order){
+        return order
```

Comment:
This can be replaced with
```java
return order
    .getOrderItems()
    .anyMatch(item -> !extractSupplierIds(item.getProduct()).contains(order.getSupplier().getId()));
```